thorlcr/activities/indexread/thindexread.cpp

```diff
@@ -73,54 +73,71 @@ class CIndexReadBase : public CMasterActivity
     }
     void prepareKey(IDistributedFile *index)
     {
-        IDistributedFile *f = index;
-        IDistributedSuperFile *super = f->querySuperFile();
+        IDistributedSuperFile *super = index->querySuperFile();
 
-        unsigned nparts = f->numParts(); // includes tlks if any, but unused in array
-        performPartLookup.ensureCapacity(nparts);
+        unsigned numParts = index->numParts();
 
-        bool checkTLKConsistency = (nullptr != super) && !localKey && (0 != (TIRsorted & indexBaseHelper->getFlags()));
-        if (nofilter)
-        {
-            while (nparts--) performPartLookup.append(true);
-            if (!checkTLKConsistency) return;
-        }
-        else
+        if (!nofilter)
```

Comment:
Not for now, but it would be much clearer to rename this boolean to filter and invert all the tests!